dependencies.yaml

```diff
@@ -0,0 +1,107 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+ files:
+   all:
+     output: conda
+     matrix:
+       cuda: ["11.5"]
+       arch: [x86_64]
+     includes:
+       - common_build
+       - cpp_build
+       - cudatoolkit
+       - doc
+       - python_build
+ channels:
+   - rapidsai
+   - rapidsai-nightly
+   - dask/label/dev
+   - conda-forge
+   - nvidia
+ dependencies:
+   cudatoolkit:
+     specific:
+       - output_types: conda
+         matrices:
+           - matrix:
+               cuda: "11.5"
+             packages:
+               - cudatoolkit=11.5
+           - matrix:
+               cuda: "11.4"
+             packages:
+               - cudatoolkit=11.4
+           - matrix:
+               cuda: "11.2"
+             packages:
+               - cudatoolkit=11.2
+           - matrix:
+               cuda: "11.0"
+             packages:
+               - cudatoolkit=11.0
+   common_build:
+     common:
+       - output_types: [conda]
+         packages:
+           - cmake>=3.23.1,!=3.25.0
+       - output_types: conda
+         packages:
+           - ninja
+           - pip
+   cpp_build:
+     common:
+       - output_types: conda
+         packages:
+           - nvcc_linux-64=11.5
```

Comment:
We'll need to split this by arch, since ARM doesn't use the same package name (it uses `nvcc_linux-aarch64`). See example here from cudf: https://github.com/rapidsai/cudf/blob/2048829ec66af1ceb7a9801702417d12ef13c77a/dependencies.yaml#L83-L94